examples/todo-mvc-react/src/Todos.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useRef } from 'react';
+import cn from 'classnames';
+import { useMachine } from '@xstate/react';
+import { useHashChange } from './useHashChange';
+import { Todo } from './Todo';
+import { todosMachine } from './todos.machine';
+
+function filterTodos(filter, todos) {
+  if (filter === 'active') {
+    return todos.filter((todo) => !todo.completed);
+  }
+
+  if (filter === 'completed') {
+    return todos.filter((todo) => todo.completed);
+  }
+
+  return todos;
+}
+
+const persistedTodosMachine = todosMachine.withConfig(
+  {
+    actions: {
+      persist: (context) => {
+        try {
+          localStorage.setItem('todos-xstate', JSON.stringify(context.todos));
+        } catch (e) {
+          console.error(e);
+        }
+      }
+    }
+  },
+  // initial state from localstorage
+  {
+    todo: 'Learn state machines',
+    todos: (() => {
+      try {
+        return JSON.parse(localStorage.getItem('todos-xstate')) || [];
+      } catch (e) {
+        console.error(e);
+        return [];
+      }
+    })(),
+    filter: 'all'
+  }
+);
+
+export function Todos() {
+  const [state, send] = useMachine(persistedTodosMachine, { devTools: true });
+
+  useHashChange(() => {
+    send({ type: 'SHOW', filter: window.location.hash.slice(2) || 'all' });
+  });
+
+  // Capture initial state of browser hash
+  useEffect(() => {
+    window.location.hash.slice(2) &&
+      send({ type: 'SHOW', filter: window.location.hash.slice(2) });
+  }, [send]);
+
+  const { todo, todos, filter } = state.context;
+
+  const numActiveTodos = todos.filter((todo) => !todo.completed).length;
+  const allCompleted = todos.length > 0 && numActiveTodos === 0;
+  const mark = !allCompleted ? 'completed' : 'active';
+  const markEvent = `MARK.${mark}`;
+  const filteredTodos = filterTodos(filter, todos);
+
+  const inputRef = useRef(null);
+  useEffect(() => {
+    if (todos.length === 0) {
+      inputRef.current.focus();
+    }
+  }, [todos]);
+
+  return (
+    <section className="todoapp" data-state={state.toStrings()}>
+      <header className="header">
+        <h1>todos</h1>
+        <input
+          ref={inputRef}
+          className="new-todo"
+          placeholder="What needs to be done?"
+          autoFocus
+          onKeyPress={(e) => {
+            if (e.key === 'Enter') {
+              send({ type: 'NEWTODO.COMMIT', value: e.target.value });
+            }
+          }}
+          onChange={(e) =>
+            send({ type: 'NEWTODO.CHANGE', value: e.target.value })
+          }
+          value={todo}
+        />
+      </header>
+
+      {!!todos.length && (
+        <>
+          <section className="main">
+            <input
+              id="toggle-all"
```

Comment:
Moved `section` inside the condition, because of the original example: https://todomvc.com/
It has sense not displaying the `toggle-all` when there is no todo.